File: app/src/main/java/wangdaye/com/geometricweather/db/converters/DateConverter.java
package wangdaye.com.geometricweather.db.converters;

import androidx.room.TypeConverter;

import java.util.Date;

/**
 * Stores {@link Date} as epoch milliseconds, matching GreenDAO's Date mapping.
 */
public class DateConverter {

    @TypeConverter
    public Date convertToEntityProperty(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public Long convertToDatabaseValue(Date date) {
        return date == null ? null : date.getTime();
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/converters/DateConverter.kt
package wangdaye.com.geometricweather.db.converters

import androidx.room.TypeConverter
import java.util.Date

/**
 * Stores [Date] as epoch milliseconds, matching GreenDAO's Date mapping.
 */
class DateConverter {

    @TypeConverter
    fun convertToEntityProperty(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun convertToDatabaseValue(date: Date?): Long? {
        return date?.time
    }
}
